src/azure-cli/azure/cli/command_modules/appservice/_params.py

```diff
@@ -1061,3 +1061,6 @@ def load_arguments(self, _):
     with self.argument_context('staticwebapp functions link') as c:
         c.argument('function_resource_id', help="Resource ID of the functionapp to link. Can be retrieved with 'az functionapp --query id'")
         c.argument('force', help="Force the function link even if the function is already linked to a static webapp. May be needed if the function was previously linked to a static webapp.")
+
+    with self.argument_context('staticwebapp enterprise-edge') as c:
+        c.argument("no_register", help="Don't try to register the Microsoft.CDN provider. Registration can be done manually with: az provider register --wait --namespace Microsoft.CDN. For more details, please review the documentation available at https://go.microsoft.com/fwlink/?linkid=2184995 .", default=False)
```

Comment:
```suggestion
        c.argument("no_register", help="Enabling enterprise-grade edge requires reregistration for the Azure Front Door Microsoft.CDN resource provider. Please register the provider with : az provider register --wait --namespace Microsoft.CDN. For more details, please review the documentation available at https://go.microsoft.com/fwlink/?linkid=2184995 .", default=False)
```

Reply:
Thanks for the suggestion, but I think this change may confuse users. The reason I wrote the help text as "Don't try to register..." is because the `--no-register` argument is a flag that, if included, makes the command not register the Microsoft.CDN RP. Also, we shouldn't tell the user to "please register the RP with <command>" -- this is just provided as an alternative if they want to do it manually. By default, the command does register the RP